### PR TITLE
chore: Upgrade installation manifests & codegen to Kustomize 3

### PR DESCRIPTION
**Pull request by jannfis**
Replacement for #3099

* Switch to Kustomize v3 for our installation manifests
* Update `argocd-test-tools` to v0.3.0

Checklist:

* [x] It's a chore, but should be documented in the Changelog
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to USERS.md.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -82,7 +82,7 @@ commands:
 jobs:
   build:
     docker:
-      - image: argoproj/argocd-test-tools:v0.2.0
+      - image: argoproj/argocd-test-tools:v0.3.0
```
Comment (jessesuen) [inline .circleci/config.yml:85]: @alexmt says we can probably remove this file completely.

**Review by mayzhang2000 — APPROVED** [review]
LGTM.

**Comment by mayzhang2000** [comment]
I am getting
```k apply -k .
error: rawResources failed to read Resources: Load from path ./application-controller failed: './application-controller' must be a file (got d='/Users/mayz985/GitHub/jannfis/argo-cd/manifests/base/application-controller')
```

**Comment by jessesuen** [comment]
> MTVL154376f69:base mayz985$ k apply -k .
error: rawResources failed to read Resources: Load from path ./application-controller failed: './application-controller' must be a file (got d='/Users/mayz985/GitHub/jannfis/argo-cd/manifests/base/application-controller')


We're okay with making kustomize 2 no longer work

File: Makefile [hunk 2]
```diff
@@ -201,7 +201,7 @@ manifests-local:
 	./hack/update-manifests.sh
 
 .PHONY: manifests
-manifests: test-tools-image
```
Comment (alexmt) [inline Makefile:204]: I don't think we need any changes in Makefile. Can you revert?

**Review by alexmt — APPROVED** [review]
LGTM

**Pull request merged**